### DISCUSSION TITLE
Virtual manuals?

### DISCUSSION THREAD

**Discussion by jason-barba** [GrandOrgue/grandorgue#1364] [Help]
I have only two manuals in my organ console. There are several interesting sample sets out there built for three (and more) manuals. Is there a simple - meaning easy to change during playing - way to allow for e.g. physical Manual II to function as physical Manual III? I.e. a  physical manual changes from being one MANUAL999 to another MANUAL999.

//Jens

**Comment by kerkovits** [comment]
Yes, there is. There are so-called coupling manuals in the panels. If you assign your MIDI input to them, you can switch on / off the built-in couplers of GO between the coupling manuals and the real ones by MIDI messages. Con: The original couplers of the ODF will not work this way, you have to use the built-in couplers of GO if you want to attach more windchests to the same manual.

By the way, if there is a III/II coupler in the ODF, it is easier to use that directly, so you can access the pipes of manual 3 from manual 2.

**Reply by jason-barba** [reply]
Thanks for your input. My GO experience is limited so I have to return it with some follow up questions.

The case with a III/II coupler I understand as straight forward.

Another case is an organ/ODF with a coupler manual. In this case I would like to quickly change man I or man II to the act as the coupler manual. As I understand from your input, this could be solved by activating a virtual coupler manual. Correct? Can this coupler manual be controlled by at stop/button or only by a MIDI input? You say that by activating a coupler manual,  couplers defined in the ODF does no longer function. Is this correctly interpreted?

I have found the coupler manuals menu and the coupler menu, but unfortunately I do not fully understand how they act together with the "master"(?) manual and the stops. The GO help doesn't help me, I admit.

//Jens

Edited 09:15 UTC

**Reply by kerkovits** [reply]
In the menu Panels->Couplers select the coupler manual to manage (eg. coupler manual 1). There is a table of pistons. Every row corresponds to a manual (either from the ODF, or other coupler manuals). If you activate the piston labeled "8" in the row of an ODF manual, the pipes should sound. Right-clicking on piston "8" configures the MIDI message of the coupler.

Now, if for example, you assigned your physical manual 2 to coupler manual 2, you have to configure two pistons "8" at Panel->Couplers->Coupler manual 2: The one on the row of manual II, and the one on the row of manual III. This way you can either:
- Switch of both pistons: Physical manual II will not sound.
- Switch on one of the pistons: Your manual II is either II or III of the virtual organ.
- Switch on both pistons: Same effect as a III/II coupler.

**Reply by jason-barba** [reply]
Aha :-) Thanks! I will try this ASAP with a three manual sample set.

But - now I return to your first posting in this discussion - by doing the above all(?) couplers defined in the ODF are disabled? Would a coupler I/P function?

**Reply by kerkovits** [reply]
Only those couplers will have no effects which act on manuals assigned this way. E.g., in the previous example, the */II and */III couplers will not work, but */I and */P will continue to work.

**Reply by jason-barba** [reply]
OK. Once again thanks!

**Reply by jason-barba** [reply]
Iam sorry but I go wrong somewhere.

I have installed Lars Palo's Piteå sample set (III/P). I attached my physical pedals and manuals I and II to the ODF's pedal, division I and II. Successful!

I then attached my physical man II to coupling manual II and checked visually that pressing a key on physical man II was indicated as pressed on the coupling manual II. I opened Coupler coupling manual II and pressed stop "8" for both the division "Positiv" and division "Svällverk" but no pipes were heard :-( (I had pulled divisional stops ;-) )

I also tried the above with physical man II and coupling manual I, but with the same negative outcome.

Any thoughts?

**Reply by larspalo** [reply]
For PiteaMHS you can just use the III/II coupler to get play the third manual from the second, no need to fiddle around with anything special.

**Reply by jason-barba** [reply]
Thanks @larspalo, but my aim was to learn and understand more about the GO functions ;-)

**Reply by larspalo** [reply]
Ok. So if you want your second manual to temporarily function only as the third you can from within GO open the coupler panel. For manual 2 you select the U.O. (unison off) coupler and for the target manual you select the 8' coupler.

**Reply by larspalo** [reply]
With a little effort with setting a couple of generals up with scope only for above mentioned couplers and storing their on state in one and their off state in another it's even possible to activate and de activate them from a single button in scoped mode. Then no other stops are affected and you can switch the second manual to instantly function as the third and then back again.

**Reply by jason-barba** [reply]
I'am sorry @larspalo , but I am obviously doing something not right. I linked my man II to also control the Coupling manual 1, not only the Svällverk (still Piteå sample set).  I the open the Coupler panel for Coupling manual 1 and tried all four 8' couplers but no pipes heard. I can see that my man II controls the Couplingh manual 1 but that's all. :-(

**Reply by kerkovits** [reply]
There is a pitfall: What seems to be coupling manual 1 is actually coupling manual 2, because actually the 'coupling pedal' is coupling manual 1. Therefore it is easy to confuse the numbers.

**Reply by larspalo** [reply]
Ok, it all depends on what you want to do, and you'll find out that there might be many ways to do it. Here I assume that you want a way to have one of your physical MIDI keyboards to function as _either_ manual 2 _or_ manual 3 at will.

If you want to use the coupling manuals version of doing this, it's important to **not** assign the actual odf (organ) manuals any MIDI input directly from the keyboards in your console. You **only** assign your physical MIDI keyboard to a coupler manual - let us say coupling manual 1 for instance. After that is done when you play on your physical MIDI keyboard you should see the signs of the coupling manual to respond to input but no other keyboard should. 

Next you go to the normal coupler panel. In the menu that flips up you open the corresponding item for the coupling manual 1 and there you can assign any coupler you wish - in your case the ones you're interested in alternating between are the 8' couplers for Positiv and Svällverk.

But again this could be done without the coupling manuals too like I described above.

**Reply by larspalo** [reply]
> There is a pitfall: What seems to be coupling manual 1 is actually coupling manual 2, because actually the 'coupling pedal' is coupling manual 1. Therefore it is easy to confuse the numbers.

What they look like is completely irrelevant, they can function for any type of manual regardless (they all have equally many keys). Just use any of them you like!

**Reply by jason-barba** [reply]
Thank you @larspalo and @kerkovits! I final got it. Now it works. I fell in the coupling manual numbering pit. Pedal=1, Man I=2, Man II= 3 ...
:-)

**Reply by kerkovits** [reply]
> With a little effort with setting a couple of generals up with scope only for above mentioned couplers […]

@larspalo How can I do this? This is completely new for me

**Reply by larspalo** [reply]
Yes, the usage of scope and scoped is one of the most difficult to grasp! :-) This method assumes that you have your physical keyboard connected to one of the manuals in the organ (no coupling manual). A process for setting up the U.O. and 8' coupler pair to function in two generals for selecting between the two manuals is:

1. Press GC or otherwise make sure no stop or coupler is active
2. In the general panel, select Scope
3. In the coupler panel for the manual you wish base things off you select the U.O. and 8' coupler pair
4. Back in the general panel push Set and then two general buttons - say 1 and 2
5. De-activate the Set and switch to Scoped
6. Again push Set and then with the U.O. and 8' coupler pair active push general 1
7. De-activate the U.O. and 8' coupler and push general 2
8. De-activate Set

If you did it correctly only the U.O and 8' coupler pair will be affected when using the general 1 and generals 2 and while it seems that both the manuals will play, only one of them will sound. Try with different stops on the manuals and you'll hear the effect. You can now even switch back to the Regular mode and the general 1 and 2 will not affect any other stops when you push them.

I'm not saying that this is a better way, it's just another way of doing it.

**Reply by kerkovits** [reply]
Thank you very much. I have always wondered, what these buttons 'scope' and 'scoped' do, I think I am now enlightened. I will try it soon. If I understand correctly, this approach can be combined with the 'coupling manual' method (by setting up a different scope for these generals). This way, only one of the manuals would show playing, so getting a better visual feedback. I am also not saying that my approach is better (it is slightly more complicated than yours, and it also kills */II couplers, while yours kills only */III), it is just a matter of taste.
